### PR TITLE
Error on delete location : AM-186

### PR DESCRIPTION
**Pull request by Choxmi**
Changed the location to the Unknown location when user delete an appointment. This will allow to delete all the locations other than the Unknown Location. (Ticket : AM-186)

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java [hunk 1]
```diff
@@ -295,7 +295,9 @@ public AppointmentBlock voidAppointmentBlock(
 			Context.getService(AppointmentService.class).voidTimeSlot(timeSlot,
 					reason);
 		}
-
+                //Set Appointment location to Unknown Location
+                Location location=new Location(1);
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java]: The unknown location will not always be with location id of 1

Comment (Choxmi) [inline api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java]: Then the only option we have is to set location to null. But for that I have to set location column a nullable. Is that okay?

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java]: Can you start by creating a failing unit test?

**Comment by Choxmi** [comment]
I'll try to choose the unknown location from the db instead of passing id.

**Comment by mogoodrich** [comment]
@Choxmi @dkayiwa sorry I didn't see the ticket earlier, but I believe this is the wrong approach--see my comment on the ticket.

**Comment by dkayiwa** [comment]
My suggested approach is on the talk discussion thread that led to this ticket.

**Comment by Choxmi** [comment]
@dkayiwa I went through the talk discussion thread again. According to your suggestion, the fix should be done  in purge method. But there isn't a way to purge an appointment using the UI. @mogoodrich also mentioned it. My suggestion is to assign the unknown location id to appointment_block on void method. Any suggestions?

**Comment by mogoodrich** [comment]
We shouldn't be changing the purge and void methods... the fix would be to change the UI so that in the appointment scheduling ui module, when  the deletes an appointment block, you call purgeAppointmentBlock instead of deleteAppointmentBlock.  This will fully remove that entry from the appointment block table, allows you to delete the location.

**Pull request closed without merging**